services/data/src/hooks/useDataQuery.ts

```diff
@@ -1,29 +1,66 @@
-import { useState, useContext, useEffect } from 'react'
+import { useState, useContext, useEffect, useCallback } from 'react'
 import { DataContext } from '../components/DataContext'
-import { QueryState, QueryMap } from '../types/Query'
+import {
+    QueryState,
+    QueryMap,
+    RefetchCallback,
+    QueryRenderInput,
+} from '../types/Query'
+import { ContextType } from '../types/Context'
 
-export const useDataQuery = (query: QueryMap): QueryState => {
+const reduceResponses = (responses: any[], names: string[]) =>
+    responses.reduce(
```

Comment:
Consider mutating and returning the same object for performance reasons?

Reply:
Thanks @janhenrikoverland !  Since this is a manually-constructed object with at most a few but usually 1 key (each key is a parallel data query) the performance implications should be negligible.  I'll change it though incase anyone gets really crazy with their query definitions (you'd need at least 100s to see performance degradation, I think)